code/modules/projectiles/projectile/special.dm

```diff
@@ -257,3 +257,84 @@
 			visible_message("<span class='danger'>\The [src] splatters a layer of web on \the [target]!</span>")
 			new /obj/effect/spider/stickyweb(target.loc)
 	..()
+
+/obj/item/projectile/beam/tungsten
+	name = "core of molten tungsten"
+	icon_state = "energy"
+	fire_sound = 'sound/weapons/emitter2.ogg'
+	pass_flags = PASSTABLE | PASSGRILLE
+	damage = 70
+	damage_type = BURN
+	check_armour = "laser"
+	light_range = 4
+	light_power = 3
+	light_color = "#3300ff"
+
+	muzzle_type = /obj/effect/projectile/tungsten/muzzle
+	tracer_type = /obj/effect/projectile/tungsten/tracer
+	impact_type = /obj/effect/projectile/tungsten/impact
+
+/obj/item/projectile/beam/tungsten/on_hit(var/atom/target, var/blocked = 0)
+	if(isliving(target))
+		var/mob/living/L = target
+		L.add_modifier(/datum/modifier/grievous_wounds, 30 SECONDS)
+		if(ishuman(L))
+			var/mob/living/carbon/human/H = L
+
+			var/target_armor = H.getarmor(def_zone, check_armour)
+			var/obj/item/organ/external/target_limb = H.get_organ(def_zone)
+
+			var/armor_special = 0
+
+			if(target_armor >= 60)
+				var/list/throw_dirs = list(1, 2, 4, 8, 5, 6, 9, 10)
+				throw_dirs -= src.dir
+				var/turf/T = get_step(H, pick(throw_dirs))
+				H.throw_at(T, 1, 1, src)
+				H.apply_damage(20, BURN, def_zone)
+				if(target_limb)
+					armor_special = 2
+					target_limb.fracture()
+
+			else if(target_armor >= 45)
+				H.apply_damage(15, BURN, def_zone)
+				if(target_limb)
+					armor_special = 1
+					target_limb.dislocate()
+
+			else if(target_armor >= 30)
+				H.apply_damage(10, BURN, def_zone)
+				if(prob(30) && target_limb)
+					armor_special = 1
+					target_limb.dislocate()
+
+			else if(target_armor >= 15)
+				H.apply_damage(5, BURN, def_zone)
+				if(prob(15) && target_limb)
+					armor_special = 1
+					target_limb.dislocate()
+
+			if(armor_special > 1)
+				target.visible_message("<span class='cult'>\The [src] slams into \the [target]'s [target_limb], reverberating loudly!</span>")
+
+			else if(armor_special)
+				target.visible_message("<span class='cult'>\The [src] slams into \the [target]'s [target_limb] with a low rumble!</span>")
+
+	..()
+
+/obj/item/projectile/beam/tungsten/on_impact(var/atom/A)
+	if(istype(A,/turf/simulated/shuttle/wall) || istype(A,/turf/simulated/wall) || (istype(A,/turf/simulated/mineral) && A.density) || istype(A,/obj/mecha) || istype(A,/obj/machinery/door))
+		var/blast_dir = src.dir
+		A.visible_message("<span class='danger'>\The [A] begins to glow!</span>")
+		spawn(2 SECONDS)
+			var/blastloc = get_step(A, blast_dir)
+			if(blastloc)
+				explosion(blastloc, -1, -1, 2, 3)
+	..()
+
+/obj/item/projectile/beam/tungsten/Bump(atom/A as mob|obj|turf|area, forced=0)

```

Comment:
mob, obj, turf, and area are all subtypes of atom, so you don't need to cast it as any of them.